pkg/git/git.go

```diff
@@ -244,3 +253,26 @@ func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
 	f.Close()
 	return true, nil
 }
+
+func validateGitAuth(logger *zap.SugaredLogger, url string) error {
+	homeenv := os.Getenv("HOME")
+	sshCred := true
+	if _, err := os.Stat(homeenv + "/.ssh"); os.IsNotExist(err) {
+		sshCred = false
+	}
+	urlSSHFormat := validateGitSSHURLFormat(url)
+	if sshCred && !urlSSHFormat {
+		logger.Warnf("The given URL does not match the ssh credential")
+	} else if !sshCred && urlSSHFormat {
+		logger.Warnf("The given ssh URL does not match the credential")
+	}
+
+	return nil
+}
+
+func validateGitSSHURLFormat(url string) bool {
```

Comment:
It would be really useful to include tests for this function, even though it's private, to make sure that we don't accidentally break any accepted SSH URL formats in future.